src/ray/gcs/pubsub/gcs_pub_sub.cc

```diff
@@ -174,5 +174,12 @@ std::string GcsPubSub::GenChannelPattern(const std::string &channel,
   return pattern.str();
 }
 
+bool GcsPubSub::IsUnsubscribed(const std::string &channel, const std::string &id) {
+  std::string pattern = GenChannelPattern(channel, id);
+
+  absl::MutexLock lock(&mutex_);
```

Comment:
Not related to the PR, but why do we need mutex here? I thought gcs server was single threaded?

Reply:
This is the GcsPubSub called by the gcs client. The GcsPubSub will have a separate thread(client_io_service_thread_) to process the unsubscribe reply, and we will query whether unsubscribe is successful in the UT main thread, so there will be multi threading problems.